python/ray/data/_internal/delegating_block_builder.py

```diff
@@ -53,7 +57,10 @@ def add_block(self, block: Block):
             return
         if self._builder is None:
             self._builder = accessor.builder()
-        self._builder.add_block(block)
+        if isinstance(accessor, ArrowBlockAccessor):
+            self._builder.add_block(accessor._table)
+        else:
+            self._builder.add_block(block)
```

Comment:
Seems like we can unify these two paths instead of special-casing for `ArrowBlockAccessor`.
```suggestion
        self._builder.add_block(accessor.to_block())
```